lib/modules/datasource/terraform-provider/index.ts

```diff
@@ -79,6 +81,30 @@ export class TerraformProviderDatasource extends TerraformDatasource {
     const serviceDiscovery = await this.getTerraformServiceDiscoveryResult(
       registryURL
     );
+    try {
+      return await this.queryRegistryExtendedApi(
+        serviceDiscovery,
+        registryURL,
+        repository
+      );
+    } catch (err) {
+      return await this.queryRegistryVersions(
```

Comment:
check if it's a 404 error and only try on those.
we should also not try this on official `registry.terraform.io`

Reply:
I wrote this at first - but it seemed to be, if this url isnt part of the spec, a third party repo could easily reply with a 301 -> to the homepage or something similarly unhelpful, also this way, if the response from that that url is malformed, and there is a TypeError, it will still be able to get the information from the other url - for the provider url in particular, the url used in the original, `queryRegistryExtendedApi`, is not documented anywhere.

Reply:
The same applies to  terraform-module  - although its documented for registry.terraform.io its not for others, and there have clearly been problems with it, I assume thats why there is a sanity check on line 99

Reply:
let's only use the old rich API requests for official registry, as there we know it's a official documented API and use the other for any other registry.

so no fallback at all and no duplicated calls

Reply:
I was only able to find documentation of the rich api for modules, not for providers - but as its been working we could keep using it.

but yes, I see sense in skipping rich for 3rd parties, and just using offical.

so then its no longer a fall back, its just a 1st party, use old, 3rd party, use new. that does seem better for number of requests.